### PR TITLE
Bug 1904523: Add useMultiNetworkPolicy flag to enable multiNetworkPolicy

### PR DESCRIPTION
**Pull request by s1061123**

### CODE REVIEW

**Comment by dougbtv** [comment]
Looks good to me, thanks Tomo!

File: operator/v1/types_network.go [hunk 1]
```diff
@@ -60,6 +60,11 @@ type NetworkSpec struct {
 	// 'false' and multiple network support is enabled.
 	DisableMultiNetwork *bool `json:"disableMultiNetwork,omitempty"`
 
+	// useMultiNetworkPolicy specifies whether or not multiNetworkPolicy
+	// should be used. If unset, this property defaults to 'false'
+	// and multiNetworkPolicy is not used.
+	UseMultiNetworkPolicy *bool `json:"useMultiNetworkPolicy,omitempty"`
```
Comment (soltysh) [inline operator/v1/types_network.go:66]: What is NetworkPolicy here? How this is different than above `DisableMultiNetwork`?

**Review by soltysh — CHANGES_REQUESTED** [review]
/hold

**Comment by dougbtv** [comment]
multi network is enabled by default, whereas this is disabled by default.

DisableMultiNetwork disables Multus CNI.

Whereas UseMultiNetworkPolicy enables a controller which allows for
NetworkPolicy-like objects to be used on additional networks as created by
Multus CNI.

**Comment by dougbtv** [comment]
/retitle Bug 1904523: Add useMultiNetworkPolicy flag to enable multiNetworkPolicy

File: operator/v1/types_network.go (hunk 1 above)
Comment (dougbtv) [reply operator/v1/types_network.go:66]: How about changing the comment to:

```
UseMultiNetworkPolicy enables a controller which allows for NetworkPolicy-like objects to be used on additional networks as created by Multus CNI
```

Comment (s1061123) [reply operator/v1/types_network.go:66]: Changed.

**Comment by s1061123** [comment]
@soltysh PTAL again?

**Comment by s1061123** [comment]
@jwforres Could you please take a look into it?

File: operator/v1/types_network.go (hunk 1 above)
Comment (soltysh) [reply operator/v1/types_network.go:66]: It's still not clear what's the difference between these two fields.

File: operator/v1/types_network.go [hunk 2]
```diff
@@ -60,6 +60,11 @@ type NetworkSpec struct {
 	// 'false' and multiple network support is enabled.
 	DisableMultiNetwork *bool `json:"disableMultiNetwork,omitempty"`
 
+	// useMultiNetworkPolicy enables a controller which allows for
+	// NetworkPolicy-like objects to be used on additional networks as
+	// created by Multus CNI
```
Comment (knobunc) [inline operator/v1/types_network.go:65]: How about:

useMultiNetworkPolicy enables a controller which allows for NetworkPolicy-like objects to be used on additional networks as created by Multus CNI.  If unset, this property defaults  to 'false' and multi-network NetworkPolicy-like objects are ignored.  If 'disableMultiNetwork' is 'true' then the value of this field is ignored.

Comment (s1061123) [reply operator/v1/types_network.go:65]: Thanks, changed.

**Review by soltysh — APPROVED** [review]
/hold cancel
/lgtm
/approve

**Pull request merged**